naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2480,14 +2508,28 @@ impl Parser {
 
         let mut lexer = Lexer::new(source);
         let mut tu = ast::TranslationUnit::default();
+        let mut enable_extensions = EnableExtensions::empty();
 
         // Parse directives.
-        #[allow(clippy::never_loop, unreachable_code)]
         while let Ok((ident, span)) = lexer.peek_ident_with_span() {
             if let Some(kind) = DirectiveKind::from_ident(ident) {
                 self.push_rule_span(Rule::Directive, &mut lexer);
                 let _ = lexer.next_ident_with_span().unwrap();
                 match kind {
+                    DirectiveKind::Enable => {
+                        enable_extensions.add(self.enable_extension_ident(&mut lexer)?);
+                        while lexer.skip(Token::Separator(',')) {
+                            enable_extensions.add(self.enable_extension_ident(&mut lexer)?);
+                        }
```

Comment:
This parsing of the comma-separated list is actually incorrect; we're not permitting a trailing comma here (CC @FL33TW00D, since this is their initial implementation). I think a human-friendly description of the parsing algorithm would be:

1. Parse a word token, and validate it as an extension we recognize. If we one of those steps fail, bail with the corresponding error.
2. Peek at input. If the next token is a comma, then consume it, and peek at the next token:
	1. If it's a word, go to step 1.
	1. If it's not a semicolon, bail with an error.
3. Parse a semicolon, or bail with an error.

I'll fix this in a bit.